src/test/java/org/jenkinsci/plugins/github/config/ConfigAsCodeTest.java

```diff
@@ -0,0 +1,96 @@
+package org.jenkinsci.plugins.github.config;
+
+import io.jenkins.plugins.casc.ConfigurationContext;
+import io.jenkins.plugins.casc.Configurator;
+import io.jenkins.plugins.casc.ConfiguratorRegistry;
+import io.jenkins.plugins.casc.misc.ConfiguredWithCode;
+import io.jenkins.plugins.casc.misc.JenkinsConfiguredWithCodeRule;
+import io.jenkins.plugins.casc.model.CNode;
+import io.jenkins.plugins.casc.model.Mapping;
+import org.junit.Rule;
+import org.junit.Test;
+
+import java.util.List;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.*;
+import static org.jenkinsci.plugins.github.test.GitHubServerConfigMatcher.*;
+
+public class ConfigAsCodeTest {
+
+    @Rule
+    public JenkinsConfiguredWithCodeRule r = new JenkinsConfiguredWithCodeRule();
+
+    @Test
+    @ConfiguredWithCode("configuration-as-code.yml")
+    public void shouldSupportConfigurationAsCode() throws Exception {
+
+        GitHubPluginConfig gitHubPluginConfig = GitHubPluginConfig.all().get(GitHubPluginConfig.class);
+
+        /** Test Global Config Properties */
+
+        assertThat(
+                "getHookUrl() is configured",
+                gitHubPluginConfig.getHookUrl().toString(),
+                is("http://some.com/github-webhook/secret-path")
+        );
+
+        assertThat(
+                "getHookSecretConfig().getCredentialsId() is configured",
+                gitHubPluginConfig.getHookSecretConfig().getCredentialsId(),
+                is("hook_secret_cred_id")
+        );
+
+        /** Test GitHub Server Configs */
+
+        assertThat("configs are loaded", gitHubPluginConfig.getConfigs(), hasSize(2));
+
+        assertThat("configs are set", gitHubPluginConfig.getConfigs(), hasItems(
+                both(withName(is("Public GitHub")))
+                        .and(withApiUrl(is("https://api.github.com")))
+                        .and(withCredsId(is("public_cred_id")))
+                        .and(withClientCacheSize(is(20)))
+                        .and(withIsManageHooks(is(true))),
+                both(withName(is("Private GitHub")))
+                        .and(withApiUrl(is("https://api.some.com")))
+                        .and(withCredsId(is("private_cred_id")))
+                        .and(withClientCacheSize(is(40)))
+                        .and(withIsManageHooks(is(false)))
+        ));
+    }
+
+    @Test
+    @ConfiguredWithCode("configuration-as-code.yml")
+    public void exportConfiguration() throws Exception {
```

Comment:
you could probably write a simpler test with a fixture, there's nothing dynamic here so should be straightforward:
https://github.com/jenkinsci/slack-plugin/blob/master/src/test/java/jenkins/plugins/slack/jcasc/ConfigurationAsCodeTest.java#L34

Reply:
What is the difference with the source yaml?

Reply:
the source yaml is the same, the expected one doesn't have the attribute that you do a .get on, in the case of slack it's missing `slackNotifier` but has all the fields under it

Reply:
i not sure also btw, final yaml is missing default return settings, is it ok?

Reply:
@timja i can merge&release finally this caasc support and i can accept PR for test update if you want to change it

Reply:
probably not worth it, it works and is unlikely to ever need to change.